### PR TITLE
Swift Package Manager Support

### PR DESCRIPTION
**Pull request by robotive**
### Main Changes

• Adds support for Swift Package Manager.

### Auxiliary Changes

• Updated Xcode project to latest/recommended settings.
• Updated project structure to match default SPM structure.
• Updated to Swift 5.
• Updated Travis config.

### Notes

• As the current version of SPM doesn't support bundling resources, unit tests won't be running via SPM as the project is using a JSON resource file. As soon as [SE-0271](https://github.com/apple/swift-evolution/blob/master/proposals/0271-package-manager-resources.md) is implemented, this issue should be fixed.
• Minor changes were done to the podspec to account for the project structure changes, but these changes were not tested as I am not using CocoaPods.
• The Readme file was also updated to add notes about SPM. They mention the version 2.1.0 which I thought would make sense to tag the changes with after they are merged into master.

```
Closes Issue #11.
```

### CODE REVIEW

**Comment by robotive** [comment]
@stphnclysmth @boronine Hi! Any chances we can get this one merged? Thanks.

**Comment by boronine** [comment]
This looks good to me, as does https://github.com/hsluv/hsluv-swift/pull/12

I'll wait for @stphnclysmth to check it out and maybe release a new version of this package. If he is busy we'll see what we can do to keep the progress going. Several of HSLuv packages are published under a project-level accounts where credentials are shared by members: https://github.com/hsluv/hsluv/tree/master/secrets

**Comment by robotive** [comment]
@boronine Thanks for looking into it! If #12 gets merged first, I might need to rebase it, although it shouldn’t be much effort.

**Comment by boronine** [comment]
I got the maintainer's permission to take over the repo but I'm going to need somebody's help here. After merging #12 there seems to be quite a few conflicts. @robotive, would you be able to fix those? If they are too hairy, can we break up the PR somehow? I don't develop for the Apple platform so I'll just merge whatever you think is good. Once that's done we can see about releasing a new version.

**Comment by robotive** [comment]
Hi @boronine, I can have a look by the end of the week and will get back to you. Good to see there is some progress on this actually 😀.

File: Package.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// swift-tools-version:5.1
+// The swift-tools-version declares the minimum version of Swift required to build this package.
+
+import PackageDescription
+
+let package = Package(
+    name: "HSLuvSwift",
+    platforms: [
+        .iOS(.v8),
+        .macOS(.v10_10),
```
Comment (vyshane) [inline Package.swift:10]: In the latest [podspec](https://github.com/hsluv/hsluv-swift/blob/master/HSLuvSwift.podspec) I see:

```
s.ios.deployment_target = '13.2'
s.osx.deployment_target = '10.11'
```

Comment (robotive) [reply Package.swift:10]: I've fixed this in the newly created PR.

**Pull request closed without merging**